### PR TITLE
Store client uptime in Redis

### PR DESCRIPTION
**Pull request by 80degreeswest**
The client (server/worker) uptime will be used for display purposes on the Buildfarm Admin dashboard.

### CODE REVIEW

**Comment by 80degreeswest** [comment]
> Make this an implementation detail of backplane.start for redis.
> Don't refer to it as a 'container' start, because it has nothing to do with a container. Backplane 'client' maybe.
> Don't use the `keyName` verbatim - add a prefix. Otherwise you will get collisions and no namespacing.

Got it!

> Set an expiry for the key entry.
> I would use the 'service' name (with the guid that is supposed to follow a single instance).
> Make the backplane refresh periodically, so that you don't a) get expiry when you don't want it, and b) can keep track of whether the thing is reporting in.

I think I'd rather keep it ip address based - we are not keeping instance IDs anywhere in BF and it's also cloud specific.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 1]
```diff
@@ -549,6 +550,10 @@ public void start() throws IOException {
     if (config.getRunFailsafeOperation()) {
       startFailsafeOperationThread();
     }
+
+    // Record client start time
+    client.call(
+        jedis -> jedis.set("uptime/" + clientPublicName, Long.toString(new Date().getTime())));
```
Comment (luxe) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:556]: I'd consider uptime the entire duration.  start_time might be less confusing and you can calculate uptime from it.

File: src/main/java/build/buildfarm/worker/shard/Worker.java [hunk 2]
```diff
@@ -786,7 +786,7 @@ public void run() {
 
   public void start() throws InterruptedException {
     try {
-      backplane.start();
+      backplane.start(config.getPublicName());
```
Comment (luxe) [inline src/main/java/build/buildfarm/worker/shard/Worker.java:789]: Not related to this diff, but I kind of wish we saved the entire config in redis so we could query anything about a worker/scheduler.  
key: ip / value: config

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 1 above)
Comment (80degreeswest) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:556]: Good point, I'll update it.

File: src/main/java/build/buildfarm/worker/shard/Worker.java (hunk 2 above)
Comment (80degreeswest) [reply src/main/java/build/buildfarm/worker/shard/Worker.java:789]: Sounds like a new story for the backlog!

**Pull request merged**